tests/php/Core/PathTest.php

```diff
@@ -47,10 +47,18 @@ public static function providerTestJoinPaths()
             [['', '/root', '/', ' ', '/', '\\'], '/root'],
             [['', 'root', '/', ' ', '/', '\\'], 'root'],
             [['\\', '', '/root', '/', ' ', '/', '\\'], '/root'],
+            [[' one ', ' ', 'two', null], 'one/two'],
```

Comment:
Added because the existing tests didn't validate why we have two `array_filter` calls.